server/src/test/java/io/crate/http/HttpTestServer.java

```diff
@@ -140,29 +141,11 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Except
         }
 
         private void handleHttpRequest(ChannelHandlerContext ctx, HttpRequest msg) throws UnsupportedEncodingException {
-            String uri = msg.uri();
-            QueryStringDecoder decoder = new QueryStringDecoder(uri);
-            logger.debug("Got Request for " + uri);
             HttpResponseStatus status = fail ? HttpResponseStatus.BAD_REQUEST : HttpResponseStatus.OK;
-
             ByteArrayOutputStream out = new ByteArrayOutputStream();
             try {
                 JsonGenerator generator = JSON_FACTORY.createGenerator(out, JsonEncoding.UTF8);
```

Comment:
Why not creating the generator at the caller side? It's not used here afaik. I'd suggest to pass the output stream instead, this would allow handlers to respond whatever content they want.

Reply:
> Then we would need to repeat JSON_FACTORY configuration code in all callers

yep, but given that it's one line, I don't see a big issue.
But also fine to keep it like that, no strong opinion.